### PR TITLE
Feature/unified nagivation

### PR DESCRIPTION
**Pull request by Dopyo**
## What does this PR do?
Added Unified Navigation that combines indoor and outdoor navigation by providing the user a simple way to step through the route instructions.

After choosing the start and end destination, the user can press "view direction" button -> then press "start" to start the navigation process.
During this process, the user will be shown the first step and the "next" button in the routeInstruction component that allows them to show the next view for the next step.
The process is finished at the end of the instruction steps when the user press the "done" button.
The user can also press the "prev" button to step back.
The routeInstruction component is collapsible to hide/expand the instructions.

image

## How to test
- [x] Unit tests: npm test
- [x] Test Coverage: npm test (or npm run test:coverage)

## Linked issue
Closes #

## Review checklist
- [x] code is readable
- [x] tests updated/added
- [x] edge cases handled

### CODE REVIEW

File: mobile/src/hooks/useNavigationState.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    const mode = transportMode === 'SHUTTLE' ? 'driving' : GOOGLE_DIRECTIONS_MODE[transportMode];
+    const params = new URLSearchParams({
+        origin: `${start.location.lat},${start.location.lng}`,
+        destination: `${destination.location.lat},${destination.location.lng}`,
+        key: googleMapsApiKey,
+        mode,
+    });
+    const url = `https://maps.googleapis.com/maps/api/directions/json?${params.toString()}`;
+    
+    try {
```
[the remark this answers is not in the document]
Comment (Dopyo) [reply mobile/src/hooks/useNavigationState.ts]: Sure. I added a defensive check using optional chaining (?.) and type validation to safely handle Google Maps API responses that may have unexpected shapes with the new commit 958c80953edfc565665d6d1e7b59552de095c058
